### PR TITLE
Result<T> improvements

### PR DESCRIPTION
**Pull request by erak**
Introduces a static member function to create errors and a merger that could be used in https://github.com/ethereum/solidity/pull/6282.

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Language Features:
 
 
 Compiler Features:
+ * Type Checker: Improves result handling of internal conversion checks.
```
Comment (erak) [inline Changelog.md]: Perhaps we don't need an entry here?

File: libdevcore/Result.h [hunk 2]
```diff
@@ -53,6 +62,16 @@ class Result
 	/// @returns the error message (can be empty).
 	std::string const& message() const { return m_message; }
 
+	/// Merges _other into this using the _merger
+	/// and appends the error messages. Meant to be called
+	/// with logical operators like logical_and, etc.
+	template<typename F>
+	void merge(Result<ResultType> const& _other, F merger)
```
Comment (Marenz) [inline libdevcore/Result.h]:
```suggestion
	void merge(Result<ResultType> const& _other, F _merger)
```

File: Changelog.md (hunk 1 above)
Comment (Marenz) [reply Changelog.md]: Yeah, I don't think so either.

**Review by Marenz — CHANGES_REQUESTED** [review]
Apart from my comment lgtm

**Comment by erak** [comment]
Updated.

File: libdevcore/Result.h [hunk 3]
```diff
@@ -54,14 +62,13 @@ class Result
...
-	/// Merges _other result into this using the _merger
-	/// and appends the error messages.
-	/// Meant to be called with logical operators like
-	/// logical_and, etc.
+	/// Merges _other into this using the _merger
+	/// and appends the error messages. Meant to be called
+	/// with logical operators like logical_and, etc.
 	template<typename F>
-	void merge(Result<ResultType> const& _other, F merger)
+	void merge(Result<ResultType> const& _other, F _merger)
```
Comment (Marenz) [inline libdevcore/Result.h:69]: You applied the fix to the wrong commit

Comment (erak) [reply libdevcore/Result.h:69]: I was lazy and you're right. I will change it.

**Review by Marenz — APPROVED** [review]

File: libdevcore/Result.h (hunk 3 above)
Comment (chriseth) [reply libdevcore/Result.h:69]: Has this been resolved?

Comment (erak) [reply libdevcore/Result.h:69]: Yes :)

**Pull request merged**